# Q-11. 
Given the data of a feature contributing to different classes

Link: https://drive.google.com/file/d/1mCjtYHiX--mMUjicuaP2gH3k-SnFxt8Y/view?usp=share_

* Check whether the distribution of all the classes are the same or not.
* Check for the equality of variance/
* Which amount LDA and QDA would perform better on this data for classification and why.
* Check the equality of mean for between all the classes.

In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import scipy.stats as stats
from warnings import filterwarnings
filterwarnings('ignore')

The daset in the given link was downloaded and saved as 'data.csv' and loaded again as df.

# Data Loading

In [2]:
df = pd.read_csv('data.csv')

# Initializing column names for a more intuitive coding
before = ' Blood Pressure Before (mmHg)'
after = ' Blood Pressure After (mmHg)'

df

Patient ID   Blood Pressure Before (mmHg)   Blood Pressure After (mmHg)
0            1                            130                           120
1            2                            142                           135
2            3                            120                           118
3            4                            135                           127
4            5                            148                           140
..         ...                            ...                           ...
95          96                            136                           129
96          97                            143                           137
97          98                            127                           123
98          99                            139                           135
99         100                            135                           130

[100 rows x 3 columns]

## Check whether the distribution of all the classes are the same or not.

There are no classes in the given data. However to check the distribution of numeric variables accross multiple classes on a target variable, we use ANOVA (Analysis of Variance) test. Which uses the f-distribution curve to ascertain weather the mean values of different classes are statistically similar or different. ANOVA test checks the squared dispersion *within* the same class and *among* two separate classes.

## Check for the equality of variance

In [4]:
statistic, p_value = stats.levene(df[before], df[after])
statistic, p_value

(0.18038002140150966, 0.6715080090945376)

We see that the p-value (0.6715) is greater than the commonly taken significance level of 5%. Therefore, **the variances are equal** in the before and the after data.

## Which amount LDA and QDA would perform better on this data for classification and why.

We cannot perform LDA (Linear discriminant analysis) or QDA (Quardratic discriminant analysis) for classification of the given dataset as these methods are for classification of categorical data, but the given data set has only numeric features, which is not suitable to apply on LDA or QDA.

## Check the equality of mean for between all the classes.

We donot have categorical features in our dataset so we cannot perform such a check. However, such a check can be performed by a one-way ANOVA test.